Sources/SwiftSyntax/Identifier.swift

```diff
@@ -12,23 +12,58 @@
 
 /// A canonicalized representation of an identifier that strips away backticks.
 public struct Identifier: Equatable, Hashable, Sendable {
-  /// The sanitized `text` of a token.
+  enum IdentifierKind: Hashable {
+    case token(raw: RawIdentifier, arena: SyntaxArenaRef)
+    case string(String)
```

Comment:
How often do you create `Identifier` from `String`. I think it might be easier to just write the name to a new `SyntaxArena` and then use it.

Reply:
I think it would be nice to just be able to reuse `RawIdentifier` for this purpose. I can't only think of the place we should store the newly created `SyntaxArena` to keep it alive. Can you think of a way to put it inside `Identifier` and remain `Hashable` and `Sendable` conformance?

Reply:
I just checked and I think that in production all usages of `Identifier(String)` use statically known strings. For that we can continue using the current `RawIdentifier` because we can initialize a `SyntaxText` from a `StaticString`

```swift
public struct RawIdentifier: Equatable, Hashable, Sendable {
  … 
  fileprivate init(_ raw: StaticString) {
    name = SyntaxText(raw)
  }
  …
}
```

And I don’t want to change the representation for tests only. Also, in tests performance does not really matter, so there we can add

```swift
fileprivate extension Identifier {
  @_spi(Testing) public init?(string: String) {
    self.init(TokenSyntax.identifier(string))
  }
}
```

Reply:
That's a good idea, thanks for helping with simplifying this!

Turns out we don't even need a `String` initializer for the tests as we can just directly use keyword tokens' `rawText` property.
```Swift
extension Identifier {
  @_spi(Testing) public init(anyToken token: TokenSyntax) {
    self.raw = RawIdentifier(token.tokenView.rawText)
    self.arena = token.raw.arenaReference
  }
}
```